mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java

```diff
@@ -190,7 +197,40 @@ public Future<JobExecution> updateJobExecutionStatus(String jobExecutionId, Stat
           }
           return promise.future();
         }, params.getTenantId())
-        .compose(jobExecution -> updateSnapshotStatus(jobExecution, params));
+        .compose(jobExecution -> updateSnapshotStatus(jobExecution, params))
+        .compose(jobExecution -> {
+          // if this composite child finished, check if all other children are finished
+          // if so, then mark the composite parent as completed
+          if (jobExecution.getSubordinationType().equals(SubordinationType.COMPOSITE_CHILD) && 
+              COMPLETE_STATUSES.contains(jobExecution.getUiStatus())) {
+            return this.getJobExecutionById(jobExecution.getParentJobId(), params.getTenantId())
+              .map(v -> v.orElseThrow(() -> new IllegalStateException("Could not find parent job execution")))
+              .compose(parentExecution -> 
+                this.getJobExecutionCollectionByParentId(parentExecution.getId(), 0, Integer.MAX_VALUE, params.getTenantId())
+                  .map(JobExecutionDtoCollection::getJobExecutions)
+                  // ensure all other children are completed
+                  .map(children ->
+                    children.stream()
+                      .filter(child -> child.getSubordinationType().equals(JobExecutionDto.SubordinationType.COMPOSITE_CHILD))
+                      .map(JobExecutionDto::getUiStatus)
```

Comment:
could you say please just for information, why the "uiStatus" field was chosen for checking the condition? Just the  "status" field seems more related to the backend.

Reply:
Either would have been acceptable; the UI status seemed to be a little more future proof in my mind since even if additional processing/parsing/etc states are added in the feature, they'd show "completed" in the UI.  Also, since the problem was that we wanted these parent jobs to show as completed in the UI when there children were, I figured the UI's view was the best basis for this

Reply:
yes, understand/agree that both fields are acceptable because, for every complete "status" field value, there is a corresponding "uiStatus" field value. Merely asked this because the "status" field is mostly used as the origin of a job status information in the module. For example as here in the method a bit above https://github.com/folio-org/mod-source-record-manager/blob/ae5cb38335d87c79e42fa9bb46186bccfa769f21/mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java#L189
Thanks for the explanation!